### ISSUE TITLE
Unified ring buffer

### ISSUE THREAD

**Issue by jfallows** [reaktivity/reaktor.java#33]
Each nukleus currently has a distinct single-reader, single-writer ring buffer for `streams` and `throttles` for each source+target nukleus combination. In practice this requires a reaktor thread to probe each ring buffer in turn in the run loop to detect new `stream` or `throttle` messages, making read ordering not guaranteed to equal write order across different ring buffers.

Instead we need to change the scope of the ring buffer to span all nuklei running on the same thread, combining `streams` and `throttles` ring buffers as well.

### Requirements
- globally unique streams ids, not just within a nukleus
- dispatch `BEGIN` frame to correct target nukleus
- detection of `stream` frames vs `throttle` frames, dispatch to corresponding handler

### Implications
- unified ring buffer created on startup
- `watchers` no longer required
- many-to-one ring buffer, multiple ring buffers (k3po)

### Layout before unification
```bash
directory/
  tcp/
    control
    streams/
      tls#tcp
  tls/
    control
    streams/
      tcp#0.0.0.0
  ...
```
First step of unification is to combine `stream` and `throttle` into single many-to-one `data` ring buffer per _nukleus_.

### Layout after per-nukleus unification
```bash
directory/
  ...
  tcp/
    control
    data
  tls/
    control
    data
  ...
```

Each ring buffer becomes the `data` queue for a specific nukleus.

### Layout after per-thread unification
```bash
directory/
  data0
  data1
  ...
  tcp/
    control
  tls/
    control
  ...
```

Each ring buffer becomes the work queue for `data` handled by specific thread, so we'll following the naming convention of `data0`, `data1`, etc for the ring buffer filenames in `directory`.

**Comment by jfallows** [comment]
The scope of each one-to-one ring buffer is now just the source.

### Layout before unification
```
directory/
  tcp/
    control
    streams/
      tls
  tls/
    control
    streams/
      tcp
  ...
```

**Comment by jfallows** [comment]
Implemented by 32d2e4cc4f46a0d019b48939d8c7a2cf2b0b1ed0
